src/main/java/com/github/jreddit/action/SubmitActions.java

```diff
@@ -216,4 +216,40 @@ else if (object.toJSONString().contains(".error.BAD_CAPTCHA.field-captcha")) {
     	
     }
     
+    /**
+     * This function edits the text of a comment or submission (self). Requires
+     * authentication.
+     *
+     * @param text The new text for the comment or selfpost
+     * @param fullname The fullname of the submission or comment
+     * @return Whether the edit succeeded
+     * @throws ActionFailedException If the action failed
+     */
+    public boolean editUserText(String fullname, String text) throws ActionFailedException {
+
+        JSONObject object = (JSONObject) restClient.post(
+                "thing_id=" + fullname + "&text=" + text + "&uh=" + user.getModhash(),
+                ApiEndpointUtils.EDITUSERTEXT,
+                user.getCookie()
+        ).getResponseObject();
+
+        if (object.toJSONString().contains(".error.USER_REQUIRED")) {
```

Comment:
you can make a local variable to keep the JSON String to not call too many times ```object.toJSONString()```. something like this:
```
String responseAsString = object.toJSONString();
````
then you can just make use of the variable declared.

Also for the returns if it was successful or not, just make a boolean variable and just set that variable if it's okay or not.
Example:
```
boolean successful = true;

if (responseAsTring.contains(".error.NOT_AUTHOR")) {
   successful = false;
}

....

return successful;
```

Not sure about the chained if's. I think we have in other places in the code where we make use of switch case. 
I see that you are making use of ```.contains()```. Does the response contain something else than just ".error.NOT_AUTHOR", for instance?